frontend/src/features/properties/components/forms/ParcelDetailForm.tsx

```diff
@@ -167,18 +176,16 @@ const ParcelDetailForm = (props: ParcelPropertyProps) => {
     if (values.pid) {
       financialErrors = validateFinancials(values.financials, 'financials', showAppraisal);
       values.buildings.forEach((building, index) => {
-        financialErrors = {
-          ...financialErrors,
-          ...validateFinancials(
-            building.financials,
-            `buildings.${index}.financials`,
-            showAppraisal,
-          ),
-        };
+        const result = validateFinancials(
+          building.financials,
+          `buildings.${index}.financials`,
+          showAppraisal,
+        );
+        _.merge(financialErrors, result);
       });
     }
 
-    const yupErrors: any = validateYupSchema(values, ParcelSchema).then(
+    const yupErrors: any = ParcelSchema.validate(values, { abortEarly: false }).then(
```

Comment:
validateYupSchema only returns the first array schema element in error, not all array elements. abortEarly on schema.validate fixes this.